app/developers/deliver_routes.py

```diff
@@ -716,6 +761,60 @@ def edit_question(
         form=wt_form,
         confirm_deletion_form=confirm_deletion_form if "delete" in request.args else None,
         managed_validation_available=get_managed_validators_by_data_type(question.data_type),
+        parent_source=parent_source,
+    )
+
+
+@developers_deliver_blueprint.route(
+    "/grants/<uuid:grant_id>/collections/<uuid:collection_id>/sections/<uuid:section_id>/forms/<uuid:form_id>/groups/<uuid:group_id>/edit",
+    methods=["GET", "POST"],
+)
+@is_platform_admin
+@auto_commit_after_request
+def edit_group(
+    grant_id: UUID, collection_id: UUID, section_id: UUID, form_id: UUID, group_id: UUID
+) -> ResponseReturnValue:
+    group = get_component_by_id(component_id=group_id)
+
+    confirm_deletion_form = ConfirmDeletionForm()
+    if "delete" in request.args:
+        try:
+            raise_if_question_has_any_dependencies(group)
+
+            if confirm_deletion_form.validate_on_submit() and confirm_deletion_form.confirm_deletion.data:
+                delete_component(group)
```

Comment:
Deleting a group that contains a question referenced elsewhere is throwing a 500 rather than showing a nice error message
![2025-08-13 13 02 07](https://github.com/user-attachments/assets/6b2639c7-03c1-4f8d-88be-bfd2ac7e616c)

Reply:
thanks very helpful, fixed this up + moving questions that depend on questions inside the group in a commit just pushing (I think that covers all combinations but we'll do another run through)